### ISSUE TITLE
 upgrade your OpenSSL library and recompile Python.

### ISSUE THREAD

**Issue by iamkishann** [Anorov/cloudflare-scrape#250]
Before creating an issue, first upgrade cfscrape with `pip install -U cfscrape` and see if you're still experiencing the problem.

## Version number

Run `pip show cfscrape` and paste the output below: 

```
Name: cfscrape
Version: 2.0.4
Summary: A simple Python module to bypass Cloudflare's anti-bot page. See https://github.com/Anorov/cloudflare-scrape for more information.
Home-page: https://github.com/Anorov/cloudflare-scrape
Author: Anorov
Author-email: anorov.vorona@gmail.com
License: UNKNOWN
Location: /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages
Requires: requests
```

## Code snippet experiencing the issue

```
        resp = scraper.get(product_url,proxies={"https":prox})
        print(resp.text)
        #sleep(15)
        soup = bs4(resp.text, 'lxml')
        sizes = []
```

## Complete exception and traceback

(*If the problem doesn't involve an exception being raised, leave this blank*)

```
Cloudflare captcha challenge presented for website.com (cfscrape cannot solve captchas)
Your OpenSSL version is lower than 1.1.1. Please upgrade your OpenSSL library and recompile Python.
```
I have captcha function that returns solved captcha token 
Mac OS x replaces openssl with LibreSSL 2.6.5
how do I link with python 
## URL of the Cloudflare-protected page

[LINK GOES HERE]

## URL of Pastebin/Gist with HTML source of protected page

[LINK GOES HERE]

**Comment by pro-src** [comment]
Hi @iamkishann,

Thanks for opening this issue.

> I have captcha function that returns solved captcha token
Mac OS x replaces openssl with LibreSSL 2.6.5
how do I link with python

I imagine that adding the LibreSSL paths to `CFLAGS` and `LDFLAGS` when compiling would be enough but it's really a stackoverflow sort of question.

I find the error message to be a little misleading since it's not necessary to recompile python on Linux or Windows. I also don't think it will be necessary on OSX after the next minor release of python.

Do you get a CAPTCHA when not using a proxy?
Do you get a CAPTCHA when using your web browser?
How often do you get a CAPTCHA when sending requests this way?
Finally, do you mind running this [script](https://gist.github.com/pro-src/17654ec3f949b0b17bd1a4aa1b4136b9) and sharing the link that it logs?

**Comment by iamkishann** [comment]
im not using proxy and get Captcha every time I try have tried with vpn and diffrent time intervals
I tried to print html file of captcha page but it gives me the "cloudflare captha presenter error" before it prints the html file 

logs from that script 
``` 
python3 /Users/Desktop/17654ec3f949b0b17bd1a4aa1b4136b9-b80206fb84dc5f74f43721847871331ae6ccd25a/report.py
Traceback (most recent call last):
  File "/Users/sarvaiya/Desktop/17654ec3f949b0b17bd1a4aa1b4136b9-b80206fb84dc5f74f43721847871331ae6ccd25a/report.py", line 112, in <module>
    scraper.get('https://pro-src.com')
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/requests/sessions.py", line 546, in get
    return self.request('GET', url, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/cfscrape/__init__.py", line 127, in request
    resp = super(CloudflareScraper, self).request(method, url, *args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/urllib3/connectionpool.py", line 601, in urlopen
    chunked=chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/urllib3/connectionpool.py", line 346, in _make_request
    self._validate_conn(conn)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/urllib3/connectionpool.py", line 850, in _validate_conn
    conn.connect()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/urllib3/connection.py", line 326, in connect
    ssl_context=context)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/urllib3/util/ssl_.py", line 329, in ssl_wrap_socket
    return context.wrap_socket(sock, server_hostname=server_hostname)
  File "/Users/sarvaiya/Desktop/17654ec3f949b0b17bd1a4aa1b4136b9-b80206fb84dc5f74f43721847871331ae6ccd25a/report.py", line 82, in wrap_socket
    cipher = ssock.cipher();
AttributeError: 'WrappedSocket' object has no attribute 'cipher'
```

**Comment by pro-src** [comment]
That is a very unusual error. I've updated the script so it will run despite the error. Please download the [script](https://gist.github.com/pro-src/17654ec3f949b0b17bd1a4aa1b4136b9) again and retry.

**Comment by iamkishann** [comment]
No CAPTCHA encountered when using :!ECDHE+SHA:!AES128-SHA
The report was saved locally as "report.md"
The report is valid Github flavored markdown, you may copy and paste it.
The dpaste link (Expires in 10 days): http://dpaste.com/2MF81XN

**Comment by pro-src** [comment]
I'm not sure why you're getting a CAPTCHA but it's specific the URL that you're targeting and/or other aspects of your code. Without the URL and exact code being used, I can advice you any further.

As a side note, you should upgrade urllib3 and requests since you're using old versions of those libraries.

**Comment by pro-src** [comment]
@iamkishann I'm going to close this issue since I don't believe that it's a problem with the library. I will continue to try to help you if you can provide the details that I mentioned or at least the URL. No worries, you can still comment and if necessary, we'll reopen the issue.

**Comment by iamkishann** [comment]
yea the site im trying keeps giving me captcha any way you could help me use the captcha function I have to get past that feel free to use my api key to test i'll change later today in that captcha function 
https://repl.it/repls/CheeryStrongParameter

**Comment by pro-src** [comment]
You can catch the error and parse the `siteKey` from the response:
```py
from cfscrape import CloudflareCaptchaError, create_scraper

scraper = create_scraper()

try:
    scraper.get('https://captcha.website')
except CloudflareCaptchaError as e:
    print(e.response.request.url)
    print(e.response.content)
```

The default headers are meant to prevent issues with CAPTCHA. I would remove those custom headers since Cloudflare's BIC is pretty picky... It cares about the order of the headers, etc.

**Comment by tede12** [comment]
I have the same error `Your OpenSSL version is lower than 1.1.1. Please upgrade your OpenSSL library and recompile Python.` I'm with Python 3.7.4 and I'm using openssl 1.0.2

**Comment by pro-src** [comment]
Please open a new issue using the bug report [template](https://github.com/Anorov/cloudflare-scrape/issues/new?assignees=&labels=bug&template=bug-report-template.md&title=).

**Comment by saadox** [comment]
im having Same issue  : 

> >>> soup = getSoup("https://animelek.net/js/all-XJLB.js") 

` Traceback (most recent call last):                                                                                       
 File "<stdin>", line 1, in <module> File "<stdin>", line 2, in getSoup   File "C:\Users\544D\AppData\Roaming\Python\Python37\site-packages\requests\sessions.py", line 546, in get   return self.request('GET', url, **kwargs)                                                                             File "C:\Users\544D\AppData\Roaming\Python\Python37\site-packages\cfscrape\__init__.py", line 128, in request             resp = self.solve_cf_challenge(resp, **kwargs)                                                                        File "C:\Users\544D\AppData\Roaming\Python\Python37\site-packages\cfscrape\__init__.py", line 194, in solve_cf_challenge                                                                                                                          redirect = self.request(method, submit_url, **cloudflare_kwargs)                                                      File "C:\Users\544D\AppData\Roaming\Python\Python37\site-packages\cfscrape\__init__.py", line 124, in request             self.handle_captcha_challenge(resp, url)                                                                              File "C:\Users\544D\AppData\Roaming\Python\Python37\site-packages\cfscrape\__init__.py", line 147, in handle_captcha_challenge                                                                                                                    raise CloudflareCaptchaError(error, response=resp)                                                                  cfscrape.CloudflareCaptchaError: Cloudflare captcha challenge presented for animelek.net (cfscrape cannot solve captchas). Your OpenSSL version is lower than 1.1.1. Please upgrade your OpenSSL library and recompile Python.  `

**Comment by pro-src** [comment]
Please open a new issue using the bug report [template](https://github.com/Anorov/cloudflare-scrape/issues/new?assignees=&labels=bug&template=bug-report-template.md&title=)
